app/lib/differential_expression_service.rb

```diff
@@ -160,7 +171,10 @@ def self.run_differential_expression_job(cluster_file, study, user, annotation_n
     params_object = DifferentialExpressionParameters.new(de_params)
     params_object.machine_type = machine_type if machine_type.present? # override :machine_type if specified
 
-    if params_object.valid?
+    if dry_run
```

Comment:
Really like this enhancement - I can see why it would be useful.  The existing validation for uniqueness would have caught this, but the error message is certainly unhelpful, and would not point you at the existing record as easily as this does.